### ISSUE TITLE
malformed MIME header line: Primary script unknown

### ISSUE THREAD

**Issue by bsacharski** [hipages/php-fpm_exporter#22]
I've wanted to add this exporter to have FPM metrics, however it would seem that either I have configured the fpm incorrectly, or the internal CGI client does not send some data.

When trying to scrape the metrics I get:
```
php-fpm-exporter_1  | time="2018-03-16T13:08:04Z" level=error msg="malformed MIME header line: Primary script unknown"
php-fpm-exporter_1  | time="2018-03-16T13:08:04Z" level=debug msg="Updated 1 pool(s) in 2.017823ms"
php-fpm-exporter_1  | time="2018-03-16T13:08:04Z" level=error msg="Error scraping PHP-FPM: malformed MIME header line: Primary script unknown"
```

My files:
api.conf (pool file)
```
[api]
user = www-data
group = www-data

listen = 127.0.0.1:9000

pm = dynamic
pm.max_children = 20
pm.start_servers = 4
pm.min_spare_servers = 2
pm.max_spare_servers = 6

pm.status_path = /fpm-status

access.log = /dev/null
```

docker.conf (another "pool" file)
```
[global]
daemonize = no
error_log = /proc/self/fd/2

[api]
; if we send this to /proc/self/fd/1, it never appears
; access.log = /proc/self/fd/2  ; stahp flooding log!

clear_env = no

; Ensure worker stdout and stderr are sent to the main error log.
catch_workers_output = yes
listen = [::]:9000
```

php.fpm.conf:
```
[global]
include=etc/php-fpm.d/*.conf
```

When I use nginx, I can see the status page correctly:
```
location ~ ^/fpm-status$ {
    access_log off;
    allow 127.0.0.1;
    allow 10.0.0.0/8;
    allow 172.16.0.0/12;
    deny all;
    fastcgi_param SCRIPT_FILENAME $document_root$fastcgi_script_name;
    include fastcgi_params;
    fastcgi_pass php:9000;
  }
```

**Comment by estahn** [comment]
Hi Sacharski,

The path to access the PHP-FPM metrics is currently hardcoded in
php-fpm_exporter to `/status`. You have it set to `/fpm-status` in your
PHP-FPM pool config.

We have a pull request open which will make it configurable. I will have a
look at it later today.

Hope that helps.

Enrico

On 17 Mar. 2018 12:09 am, "Sacharski Bartłomiej" <notifications@github.com>
wrote:

I've wanted to add this exporter to have FPM metrics, however it would seem
that either I have configured the fpm incorrectly, or the internal CGI
client does not send some data.

When trying to scrape the metrics I get:

php-fpm-exporter_1  | time="2018-03-16T13:08:04Z" level=error
msg="malformed MIME header line: Primary script unknown"
php-fpm-exporter_1  | time="2018-03-16T13:08:04Z" level=debug
msg="Updated 1 pool(s) in 2.017823ms"
php-fpm-exporter_1  | time="2018-03-16T13:08:04Z" level=error
msg="Error scraping PHP-FPM: malformed MIME header line: Primary
script unknown"

My files:
api.conf (pool file)

[api]
user = www-data
group = www-data

listen = 127.0.0.1:9000

pm = dynamic
pm.max_children = 20
pm.start_servers = 4
pm.min_spare_servers = 2
pm.max_spare_servers = 6

pm.status_path = /fpm-status

access.log = /dev/null

docker.conf (another "pool" file)

[global]
daemonize = no
error_log = /proc/self/fd/2

[api]
; if we send this to /proc/self/fd/1, it never appears
; access.log = /proc/self/fd/2  ; stahp flooding log!

clear_env = no

; Ensure worker stdout and stderr are sent to the main error log.
catch_workers_output = yes
listen = [::]:9000

php.fpm.conf:

[global]
include=etc/php-fpm.d/*.conf

When I use nginx, I can see the status page correctly:

location ~ ^/fpm-status$ {
    access_log off;
    allow 127.0.0.1;
    allow 10.0.0.0/8;
    allow 172.16.0.0/12;
    deny all;
    fastcgi_param SCRIPT_FILENAME $document_root$fastcgi_script_name;
    include fastcgi_params;
    fastcgi_pass php:9000;
  }

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/hipages/php-fpm_exporter/issues/22>, or mute the thread

**Comment by estahn** [comment]
@bsacharski This issue has been fixed with **v0.5.0**
